### PR TITLE
add overlaybd-resize tool for userspace ext4 resize on overlaybd images

### PR DESCRIPTION
**Pull request by OsentryO**
## Summary

Add `overlaybd-resize` tool that performs ext4 filesystem resize (shrink/expand) on overlaybd images in userspace, without kernel block device.

## Changes

- **New tool**: `src/tools/overlaybd-resize.cpp` — CLI tool that resizes ext4 filesystem on existing writable layer + updates vsize
- **Tests**: `src/test/resize_test.cpp` — 10 unit tests covering shrink/expand/relocation/edge cases
- **CMake integration**:
  - `CMake/Findphoton.cmake`: point to upstream release/0.6 (commit 0178d14) which includes `resize_extfs` backport (alibaba/PhotonLibOS#1276)
  - `CMake/Finde2fs.cmake`: point to merged PR data-accelerator/e2fsprogs#10, add `E2FS_RESIZE_DIR` and `E2FS_INSTALL_LIB_DIR` variables
  - `src/tools/CMakeLists.txt`: add `overlaybd-resize` build target (gated by `PHOTON_ENABLE_RESIZE`)
  - `src/test/CMakeLists.txt`: add `resize_test` build target

## Usage

```bash
overlaybd-resize --config /path/to/config.v1.json --size 8 [--service_config_path PATH] [--verbose]
```

Size is specified in GB (e.g., `8` for 8GB, `500` for 500GB). Only integer values are accepted.

**Requirements**:
- Config must have an existing `upper` layer (data + index files)
- Upper layer files must be accessible (read/write)

## Design

Operates on existing writable layer (upper required):
- **Expand**: ftruncate data file to target size → update index header (if non-sparse) → update data header → open image → resize_extfs
- **Shrink**: open image → resize_extfs → ftruncate data file → update headers

**Key features**:
- Inlined ImageService calls to avoid `exit(-1)` in `create_overlaybd`, enabling proper error handling and rollback
- Header update order: ftruncate → index → data (data written last as commit point for retry safety)
- Sparse layer detection: automatically skips index header update for sparse layers (tiny/empty index files)

## Dependencies

- Requires photon with `PHOTON_ENABLE_RESIZE=ON` (backported to release/0.6 in PR alibaba/PhotonLibOS#1276)
- Requires e2fsprogs with resize optimization (merged in PR data-accelerator/e2fsprogs#10)

## Testing

- 10 unit tests covering: basic shrink/expand, relocation, too-small target, non-aligned groups, empty disk, multiple resizes
- Verified on real overlaybd images: wordpress (no lazyinit), swebench (lazyinit), stable-diffusion (30 layers)
  - Shrink 256G→8G: 0.2-0.8s
  - Expand 256G→500G: 0.6-0.7s

### CODE REVIEW

File: src/tools/overlaybd-resize.cpp [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+    return tmp_config;
+}
+
+int main(int argc, char **argv) {
+    std::string config_path, size_str, data_dir, service_config_path;
+    bool verbose = false;
+
+    CLI::App app{"overlaybd-resize: resize ext4 filesystem on overlaybd image in userspace"};
+    app.add_option("--config", config_path,
+                   "overlaybd image config (config.v1.json from rpull)")
```
Comment (BigVan) [inline src/tools/overlaybd-resize.cpp]: remove'rpull'

File: src/tools/overlaybd-resize.cpp [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+        ->type_name("FILEPATH")
+        ->check(CLI::ExistingFile)
+        ->required();
+    app.add_option("--size", size_str,
+                   "target size (e.g. 8G, 500G, 8192M)")
+        ->required();
+    app.add_option("--data_dir", data_dir,
```
Comment (BigVan) [inline src/tools/overlaybd-resize.cpp]: remove this...
do resize on the default 'upper' in config.v1.json
